### PR TITLE
Add `dtype` argument to all losses and export `CTC`

### PR DESCRIPTION
**Pull request by james77777778**
This PR allows users to set the `dtype` for `Loss`. Previously, it was only configurable via `keras.backend.set_floatx()`.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
